src/backend/tests/it/get_users.rs

```diff
@@ -0,0 +1,186 @@
+use std::time::{Duration, UNIX_EPOCH};
+
+use crate::utils::{
+    mock::{CALLER, ISSUER_CANISTER_ID, VC_HOLDER, VP_JWT},
+    pocketic::{query_call, setup, update_call},
+};
+use candid::Principal;
+use ic_verifiable_credentials::issuer_api::CredentialSpec;
+use pocket_ic::PocketIc;
+use shared::types::user_profile::{
+    AddUserCredentialError, AddUserCredentialRequest, GetUsersRequest, GetUsersResponse, OisyUser,
+    UserProfile,
+};
+
```

Comment:
All tests have positive assertions. Can we add some that test errors?
Notably, not being able to call get_users for an unknown caller.

Maybe I missed it, but is there also a test that returns fewer users than matches_max_length and a test that returns fewer users than the number of users in memory?

Reply:
* Not being able to call for not allowe caller -> Will do!
* returns fewer users than matches_max_length -> Will do!
* returns fewer users than the number of users in memory -> Yes, `test_list_users_returns_requested_users_count` and `test_list_users_returns_filtered_users_by_updated`.